DuckDuckGo/Suggestions/ViewModel/SuggestionContainerViewModel.swift

```diff
@@ -181,4 +181,13 @@ final class SuggestionContainerViewModel {
         select(at: newIndex)
     }
 
+    func removeSuggestionFromResult(suggestion: Suggestion) {
+        suggestionContainer.result?.topHits = suggestionContainer.result?.topHits.filter({
+            !($0 == suggestion && $0.isHistoryEntry)
+        }) ?? []
+        suggestionContainer.result?.localSuggestions = suggestionContainer.result?.localSuggestions.filter({
+            !($0 == suggestion && $0.isHistoryEntry)
+        }) ?? []
```

Comment:
As mentioned in https://github.com/duckduckgo/BrowserServicesKit/pull/1027/files#r1824293881 - it'd be better to re-construct the result using the SuggestionResult's constructor than make this struct mutable.  The main reason I think is that we've written that code in such a way that something could be subscribed to changes in the container (or might be in the future) and these changes wouldn't get reflected.

Reply:
Yeah, agree with you 👍 Even though the event gets fired when the struct is mutated, it's better this way